app/services/proforma_service/convert_exercise_to_task.rb

```diff
@@ -77,62 +77,85 @@ def uuid
     end
 
     def model_solutions
-      @exercise.files.filter {|file| file.role == 'reference_implementation' }.map do |file|
+      @exercise.files.filter {|file| file.role == 'reference_implementation' }.group_by {|file| file.xml_id_path&.split('/')&.first || "co-ms-#{file.id}" }.map do |xml_id, files|
         ProformaXML::ModelSolution.new(
-          id: "ms-#{file.id}",
-          files: model_solution_file(file)
+          id: xml_id || "ms-#{files.first.id}",
+          files: files.map {|file| model_solution_file(file) }
         )
       end
     end
 
     def model_solution_file(file)
-      [
-        task_file(file).tap do |ms_file|
-          ms_file.used_by_grader = false
-          ms_file.usage_by_lms = 'display'
-        end,
-      ]
+      task_file(file).tap do |ms_file|
+        ms_file.used_by_grader = false
+        ms_file.usage_by_lms = 'display'
+      end
     end
 
     def tests
-      @exercise.files.filter(&:teacher_defined_assessment?).map do |file|
+      @exercise.files.filter(&:teacher_defined_assessment?).group_by {|file| xml_id_from_file(file).first }.map do |xml_id, files|
```

Comment:
Can you just assist me in understanding why you're grouping here (and for model solutions, but not files)? Is this related to the duplication we do in case of a re-used XML ID (like a file that is referenced multiple times)?

Reply:
Okay, I see. Then, this is only "working" (or required) when reexporting a previously imported exercise. I've tested this scenario and can confirm it is working as you intended.